Header header
string location_str
string[] alert_classes_list
